oxidauth-http/src/server/api/v1/auth/oauth2/callback.rs

```diff
@@ -0,0 +1,103 @@
+use axum::{
+    extract::{Path, Query, State},
+    response::{IntoResponse, Redirect},
+};
+use serde::{Deserialize, Serialize};
+use tracing::info;
+use url::Url;
+use uuid::Uuid;
+
+use oxidauth_kernel::{
+    JsonValue,
+    auth::authenticate_or_register::{
+        AuthenticateOrRegisterParams, AuthenticateOrRegisterService, OAuth2AuthenticateParams,
+        OAuth2AuthenticatePathParams,
+    },
+};
+
+use crate::provider::Provider;
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct PathParams {
+    pub client_key: Uuid,
+}
+
+#[tracing::instrument(name = "oauth2_authenticate_handler", skip(provider))]
+pub async fn handle(
+    State(provider): State<Provider>,
+    Path(path_params): Path<PathParams>,
+    auth_response: Query<OAuth2AuthenticatePathParams>,
```

Comment:
I feel like this is missing?

```suggestion
    Query(auth_response): Query<OAuth2AuthenticatePathParams>,
```

Reply:
It... seems to function either way?  Don't fully understand what its doing -- adding your suggestion tho